v2/spanner-common/src/main/java/com/google/cloud/teleport/v2/spanner/migrations/avro/GenericRecordTypeConvertor.java

```diff
@@ -252,15 +253,20 @@ private Map<String, Object> genericRecordToMap(GenericRecord record) {
       switch (fieldType) {
```

Comment:
Summary of changes around here:

When we covered these lines in UT, Boolean was causing a crash which got fixed.
Also the null checks were unreachable due to the already present null check on top.

This PR has UT coverage for all of the types in this switch case along with UT for null values.